examples/evm/call-contract-with-token/tests/call-contract-with-token.test.js

```diff
@@ -0,0 +1,222 @@
+    let aUSDCAvalanche;
+
+    before(async () => {
+        // Initialize a Polygon network
+        polygon = await createNetwork({
+            name: 'Polygon',
+        });
+
+        // Initialize an Avalanche network
+        avalanche = await createNetwork({
+            name: 'Avalanche',
+        });
+
+        // Extract user wallets for both networks
+        [polygonUserWallet] = polygon.userWallets;
+        [avalancheUserWallet, avalancheUserWalletTwo] = avalanche.userWallets;
+
+        // Deploy USDC token on the Polygon network
+        await polygon.deployToken('USDC', 'aUSDC', 6, BigInt(100_000e6));
+
+        // Deploy USDC token on the Fantom network
+        await avalanche.deployToken('USDC', 'aUSDC', 6, BigInt(100_000e6));
+
+        // Get token contracts for both chains
+        aUSDCPolygon = await polygon.getTokenContract('aUSDC');
+        aUSDCAvalanche = await avalanche.getTokenContract('aUSDC');
+
+        await polygon.giveToken(polygonUserWallet.address, 'aUSDC', BigInt(100e6));
+    });
+
+    describe('src chain', async () => {
+        beforeEach(async () => {
+            deployedContractPolygon = await deployContract(polygonUserWallet, CallContractWithToken, [
+                polygon.gateway.address,
+                polygon.gasService.address,
+            ]);
+            deployedContractAvalanche = await deployContract(avalancheUserWallet, CallContractWithToken, [
+                avalanche.gateway.address,
+                avalanche.gasService.address,
+            ]);
+            await aUSDCPolygon.connect(polygonUserWallet).approve(deployedContractPolygon.address, (100e18).toString());
+        });
+        afterEach(async () => {
+            await relay();
+        });
+        it('should set correct gateway and gas service addresses on src chain', async () => {
+            expect(await deployedContractPolygon.gateway()).to.equal(polygon.gateway.address);
+            expect(await deployedContractPolygon.gasService()).to.equal(polygon.gasService.address);
+        });
+        it('should deduct funds from msg.sender', async () => {
+            const totalSupplyBefore = await aUSDCPolygon.totalSupply();
+            const myBalanceBefore = await aUSDCPolygon.balanceOf(polygonUserWallet.address);
+
+            await deployedContractPolygon.sendToMany(
+                avalanche.name,
+                deployedContractAvalanche.address,
+                [avalancheUserWallet.address, avalancheUserWalletTwo.address],
+                'aUSDC',
+                6e6,
+                {
+                    value: (1e18).toString(),
+                },
+            );
+
+            const myBalanceAfter = await aUSDCPolygon.balanceOf(polygonUserWallet.address);
+            const totalSupplyAfter = await aUSDCPolygon.totalSupply();
+
+            expect(myBalanceAfter).to.equal(myBalanceBefore - 6e6);
+            expect(totalSupplyAfter).to.equal(totalSupplyBefore - 6e6); //token was burnt on src
```

Comment:
While technically correct, should we just say it was handled/removed by the gateway because the same is true if it's lock/unlock?

Reply:
sure, can change to this "token was removed by the gateway on the src chain"